cpp/src/io/parquet/page_enc.cu

```diff
@@ -586,10 +584,12 @@ inline __device__ void PackLiteralsRoundRobin(
 {
   // Scratch space to temporarily write to. Needed because we will use atomics to write 32 bit
   // words but the destination mem may not be a multiple of 4 bytes.
-  // TODO (dm): This assumes blockdim = 128 and max bits per value = 16. Reduce magic numbers.
-  // To allow up to 24 bit this needs to be sized at 96 words.
-  __shared__ uint32_t scratch[64];
-  if (t < 64) { scratch[t] = 0; }
+  // TODO (dm): This assumes blockdim = 128. Reduce magic numbers.
+  constexpr uint32_t NUM_THREADS  = 128;  // this needs to match gpuEncodePages block_size parameter
+  constexpr uint32_t NUM_BYTES    = (NUM_THREADS * MAX_DICT_BITS) >> 3;
+  constexpr uint32_t SCRATCH_SIZE = NUM_BYTES / sizeof(uint32_t);
+  __shared__ uint32_t scratch[SCRATCH_SIZE];
+  if (t < SCRATCH_SIZE) { scratch[t] = 0; }
```

Comment:
hm, in theory this should be a loop to cover the future possibility of `NUM_THREADS < SCRATCH_SIZE`.

Reply:
Good catch.  That was Devavret being clever I think 😉 There are more magic numbers at the end of this function too.  Fix those now or do a new PR?

Reply:
Up to you, the PR is already approved :D
Code freeze for 22.10 is on 29th, so we got sereval work days for any additional changes in this PR.

Reply:
Fixed.